lib/utils.js

```diff
@@ -35,4 +35,42 @@ function isStream (stream) {
   return stream !== null && typeof stream === 'object' && typeof stream.pipe === 'function'
 }
 
-module.exports = { isGzip, isDeflate, isStream }
+// It is a helper used to provide a async iteratable for
+// Readable.from
+async function * intoAsyncIterator (payload) {
+  const isBuffer = Buffer.isBuffer(payload)
+
+  if (
+    (
+    // ArrayBuffer
+      payload instanceof ArrayBuffer ||
+    // NodeJS.TypedArray
+    ArrayBuffer.isView(payload)
+    ) &&
+    // Exclude Buffer to prevent double cast
+    !isBuffer
+  ) {
+    payload = Buffer.from(payload)
+  }
+
+  // Iterator
+  if (typeof payload === 'object' && typeof payload[Symbol.iterator] === 'function' && !isBuffer) {
```

Comment:
Mayba add fast path if Buffer...

Reply:
It doesn't provides any goods but one more branch and `===` check.